prow/jira/jira.go

```diff
@@ -259,8 +288,275 @@ func (jc *client) UpdateRemoteLink(id string, link *jira.RemoteLink) error {
 	return nil
 }
 
-func (jc *client) JiraURL() string {
-	return jc.url
+func (jc *client) DeleteRemoteLink(issueID string, linkID int) error {
+	apiEndpoint := fmt.Sprintf("/rest/api/2/issue/%s/remotelink/%d", issueID, linkID)
+	req, err := jc.upstream.NewRequest("DELETE", apiEndpoint, nil)
+	if err != nil {
+		return err
+	}
+
+	// the response should be empty if it is not an error
+	emptyStruct := struct{}{}
+	resp, err := jc.upstream.Do(req, &emptyStruct)
```

Comment:
```suggestion
	resp, err := jc.upstream.Do(req, nil)
	if resp != nil { defer resp.Body.Close() }
```

Response bodies must always be closed to avoid leaking tcp connections: https://pkg.go.dev/net/http#Response

And also, if you pass a non-nil second argument, the jira client will read the body into that argument. This means the `JiraError` can not do that anymore, as it is already read but we want it to be able to do that to allow understanding errors